### PR TITLE
Mocha

### PR DESCRIPTION
**Pull request by vanceza**
Switch tests to run with the 'mocha' command line tool.
Tests are unchanged, but testcase.coffee could use some more work in the future.

### CODE REVIEW

File: tests/testcase.coffee [hunk 1]
```diff
@@ -25,11 +25,22 @@ class TestCase
     keyBindings = new KeyBindings @settings
     @keyhandler = new KeyHandler @view, keyBindings
     @register = @view.register
+    @name = options.name || "an anonymous test"
+    unless callback then throw "No callback in test"
+    if it?
```
Comment (WuTheFWasThat) [inline tests/testcase.coffee]: why an if?  so you can run without mocha?

Comment (WuTheFWasThat) [inline tests/testcase.coffee]: i'm guessing that's also why you wouldn't just do

it 'describe the test, () ->
  t = new TestCase [ ... ]
  t.sendKeys ... 

is that right?

File: package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
   },
   "scripts": {
     "start": "NODE_ENV=development nodemon -e coffee,jade,sass server.coffee",
-    "test": "./tests/run.coffee"
+    "test": "mocha --compilers coffee:coffee-script tests/tests/*"
   },
```
Comment (WuTheFWasThat) [inline package.json]: i'm guessing we also want


    "devDependencies": {
      "mocha": "latest"
    }

File: tests/testcase.coffee (hunk 1 above)
Comment (vanceza) [inline tests/testcase.coffee]: Not so much so we can run without mocha, as so we can switch from mocha to something else more easily.

Comment (vanceza) [inline tests/testcase.coffee]: Actually thought I took `if it?` out. That was the intention, but I decided it would be simpler without that line.

**Comment by WuTheFWasThat** [comment]
merged

**Pull request closed without merging**